app/controllers/ticket_purchases_controller.rb

```diff
@@ -30,6 +31,18 @@ def index
 
   private
 
+  def count_purchased_registration_tickets
+    purchases = params[:tickets][0]
+    purchased_registration_tickets = 0
+    @conference.tickets.for_registration.each do |registration_ticket|
+      purchased_registration_tickets += purchases[registration_ticket.id.to_s].to_i
+    end
```

Comment:
`Ticket.find(purchased_tickets).where(registration_ticket: true)`
If I am right here you are trying to find the purchased registration tickets but the problem is these tickets are not created yet. here purchased_tickets will be hash containing ticket id as key and quantity as the key (bought in that single purchase).